### PR TITLE
config: add no_proxy configuration

### PR DESCRIPTION
**Pull request by s-urbaniak**
Currently there is no possibility to add a NO_PROXY configuration
to the operator.

This fixes it and applies the HTTP configuration to the oauth proxy
sidecar for alertmanager.

Fixes MON-255

/cc @squat

### CODE REVIEW

File: pkg/manifests/manifests_test.go [hunk 1]
```diff
@@ -79,6 +79,7 @@ func TestUnconfiguredManifests(t *testing.T) {
 	}
 
 	_, err = f.NodeExporterServiceMonitor()
+
```
Comment (squat) [inline pkg/manifests/manifests_test.go]: nit: this newline seems accidental

**Comment by squat** [comment]
looks good overall. one small nit!

**Comment by s-urbaniak** [comment]
@squat fixed PTAL

**Pull request merged**